src/gui_tool/gui/tinker_subuis/cashe_manager.py

```diff
@@ -1,48 +1,84 @@
 import json
 import os.path as path
+from typing import List
 
 class ListCacheManager(object):
+    FIELD_DELIMITER = "\n"
+    SUB_FIELD_PREFIX = "  "
     def __init__(self, name, capacity):
         self.name = name
         self.capacity = capacity
         self.data = []
+        self.sub_fields = {}
         self.__file_name = path.join(path.dirname(path.abspath(__file__)), "store_" + self.name + ".cache")
         self.retrieve()
 
 
     def retrieve(self):
+        def parse(str_data):
```

Comment:
doesn't really matter but maybe we should just write the file as a json/yaml file. That way we dont need to specify our own parser and encoder